src/main/java/com/example/videotecha/mapper/ReservationMapper.java

```diff
@@ -0,0 +1,17 @@
+package com.example.videotecha.mapper;
+
+import com.example.videotecha.dto.ReservationDto;
+import com.example.videotecha.model.Reservation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class ReservationMapper {
+
+    public static List<ReservationDto> reservationsToReservationDtos(List<Reservation> reservations) {
+        List<ReservationDto> reservationDtos = new ArrayList<>();
+        reservations.forEach(r -> reservationDtos.add(new ReservationDto(r)));
+
+        return reservationDtos;
```

Comment:
Stream API?

Reply:
Also if you are using a predefined arraylist, make sure to pass the size arg to avoid dynamic resizing during adding.